modules/core/src/main/java/org/apache/synapse/mediators/builtin/ForEachMediator.java

```diff
@@ -96,85 +105,140 @@ public boolean mediate(MessageContext synCtx) {
 
         if (synLog.isTraceOrDebugEnabled()) {
             synLog.traceOrDebug("Saved original message property : " + originalMsgPropName +
-                                " = " + originalEnvelope +
-                                "Initialized foreach counter property, " + counterPropName +
-                                " = " + msgCounter);
+                    " = " + originalEnvelope +
+                    "Initialized foreach counter property, " + counterPropName +
+                    " = " + msgCounter);
         }
 
-        SOAPEnvelope processingEnvelope = synCtx.getEnvelope();
+        if (expression != null && expression instanceof SynapseJsonPath) {
```

Comment:
We can remove this unnecessary null check. instance of returns false if the expression is null.

Reply:
This was added to prevent null expressions getting inside the else block. If we remove null check, null expressions will get executed inside the else block.

Reply:
Still, a null expression would go inside the else block, right? This will evauluate to true only if the expression is not null.